exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,213 @@
+package com.google.cloud.opentelemetry.metric;
+
+import static io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type.MONOTONIC_DOUBLE;
+import static io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type.MONOTONIC_LONG;
+import static io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type.NON_MONOTONIC_DOUBLE;
+import static io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type.NON_MONOTONIC_LONG;
+import static java.util.logging.Level.WARNING;
+
+import com.google.api.LabelDescriptor;
+import com.google.api.Metric;
+import com.google.api.MetricDescriptor;
+import com.google.api.MonitoredResource;
+import com.google.cloud.opentelemetry.metric.MetricExporter.MetricWithLabels;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.primitives.Longs;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.Point.Builder;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.Timestamp;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.ReadableAttributes;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type;
+import io.opentelemetry.sdk.resources.Resource;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class MetricTranslator {
+
+  private static final Logger logger = Logger.getLogger(MetricTranslator.class.getName());
+
+  static final String DESCRIPTOR_TYPE_URL = "custom.googleapis.com/OpenTelemetry/";
+  static final String UNIQUE_IDENTIFIER_KEY = "opentelemetry_id";
+  static final long NANO_PER_SECOND = (long) 1e9;
+
+  static final Set<Type> GAUGE_TYPES = ImmutableSet.of(NON_MONOTONIC_LONG, NON_MONOTONIC_DOUBLE);
+  static final Set<MetricData.Descriptor.Type> CUMULATIVE_TYPES = ImmutableSet
+      .of(MONOTONIC_LONG, MONOTONIC_DOUBLE);
+  static final Set<MetricData.Descriptor.Type> LONG_TYPES = ImmutableSet.of(NON_MONOTONIC_LONG, MONOTONIC_LONG);
+  static final Set<MetricData.Descriptor.Type> DOUBLE_TYPES = ImmutableSet
+      .of(NON_MONOTONIC_DOUBLE, MONOTONIC_DOUBLE);
+
+  static final Map<String, Map<String, String>> OTEL_TO_GCP_LABELS = ImmutableMap.<String, Map<String, String>>builder()
+      .put("gce_instance", ImmutableMap.<String, String>builder()
+          .put("host.id", "instance_id")
```

Comment:
nit: Would be nice define keys as constants. Fine to fix later.
Also, ideally they should be defined in the SDK, since they are part of the [Semantic Conventions spec](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions), but that's fine for now.